### ISSUE TITLE
how can I handle the blocked event when opening a database?

### ISSUE THREAD

**Issue by retorquere** [MockingMagician/promised-db#3]
I see no handler for blocked like there is for upgradeneeded

**Comment by MockingMagician** [comment]
Hello @retorquere 

This is automatically handled when you have requested a higher database version, it tries to run the higher migration version via the `upgradeneeded` listener needed when opening the database.

https://github.com/MockingMagician/promised-db/blob/main/src/database-factory.ts#L73

**Comment by MockingMagician** [comment]
Oops I misread your question

**Comment by MockingMagician** [comment]
Indeed, there's no mechanism in place to deal with the possibility of trying to reopen the same db with a higher version when using it.

I can add its implementation with an available handler and if you have a suggestion for its automation by default, I'd love to hear it :-)

By the way, I'm wondering about the possible cases where this could happen.

**Comment by retorquere** [comment]
I think I was just confused about how this works - I bumped into it while trying to delete a database, turns out I just have to close it before I do.

**Comment by MockingMagician** [comment]
Hello @retorquere 

I decided to take your case into account and add a handler for the “blocked” event.

You can see the tests here https://github.com/MockingMagician/promised-db/blob/dd7ae8fd269a889b194f48683e012260d7eaa7d5/test/database-factory.test.ts#L68-L81 and the implementation here https://github.com/MockingMagician/promised-db/blob/dd7ae8fd269a889b194f48683e012260d7eaa7d5/src/database-factory.ts#L73-L80

Available in version v2.6.1

If you see any use cases where it would be necessary for the user to take control of this handler, let me know so that I can implement in the API the possibility of passing a custom handler written by the user.

**Comment by MockingMagician** [comment]
Improvement:

Add a user-customizable handler to perform custom actions in this type of event.

Example:

- Warn other tabs that a DB update is required and therefore close the database in all tabs.
- Warn user
- Or any other case required by an application with its own needs.

**Comment by MockingMagician** [comment]
Custom user handler available in v4.0.0
